### PR TITLE
Skills updates

### LINKED ISSUE #8648: [Bug]: Chatbot produces error: Can't select from airline_passenger_satisfaction in project: unkown
### Short description of current behavior

When querying chatbot, it produces an error `Error: Can't select from airline_passenger_satisfaction in project: unkown`

Slack thread: https://mindsdbcommunity.slack.com/archives/C01S2T35H18/p1703167276710179

### LINKED ISSUE #8764: [Bug]: Knowledge base with default embedding mode does't work in chatbot
### Short description of current behavior

Default embedding for KB is huggingface embeddings but  emedding model which [created inside of skill](https://github.com/mindsdb/mindsdb/blob/e45ca9fea349a999be6b5de098bb3400a088b0aa/mindsdb/integrations/handlers/langchain_handler/skill_tool.py#L58) is openai model.
It leads error:
`Embedding dimension 1536 does not match collection dimensionality 384...`

### Video or screenshots

_No response_

### Expected behavior

_No response_

### How to reproduce the error
```sql
CREATE KNOWLEDGE BASE my_knowledge_data;

INSERT INTO my_knowledge_data(content)
VALUES ('I drink tea.');

CREATE SKILL skill_from_data
USING
    type = 'knowledge_base',
    source = 'my_knowledge_data',
    description = 'My data';

CREATE MODEL my_gpt_slack_model
PREDICT answer
USING
    engine = 'langchain_engine',
    input_column = 'question',
    api_key = '<>',
    model_name = 'gpt-3.5-turbo-16k',
    mode = 'conversational',
    user_column = 'question' ,
    assistant_column = 'answer',
    temperature=0,
    verbose=True,
    prompt_template='Answer the user input in a helpful way';

CREATE AGENT super_agent
USING
   model = 'my_gpt_slack_model',
   skills = ['skill_from_data'];

CREATE CHATBOT demo2_chatbot USING
  database = "mindsdb_slack",
  agent = "super_agent";
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ea-rus**
## Description

Changes:
- anyscale handler support in langchain handler, parameters:
  - provider='anyscale' 
  - model_name - anyscale model
  - anyscale_api_key - anyscale api key
  - base_url - url to api, default is "https://api.endpoints.anyscale.com/v1",
- litellm support in langchain handler, parameters:
  - provider='litellm' 
  - model_name - model created in mindsdb
  - litellm_api_key - api key created in mindsdb
  - base_url - url to mindsdb instance 

Fixes:
- Chatbot fixes:
  - don't read history before start of chatbot.
- Text2sql fixes:
  - check tables in query before execution. if table is not found : propose correct list of tables
  - custom tool instead of using SQLDatabaseToolkit:
     - provide only one function to call and list of tables with list of columns
  - if query response is too big:
    -  show only 30 first rows and count of rows
    -  provide pd.describe for whole date
  - don't use standard tools if skills are defined
- KB fixes:
  -  If embedding model and vector db have been created automatically: remove them when KB is removed
  - call knowledge base directly from skill (used to call RAG handler)  fix #8764
- Langchain fixes
  - removed python_repl tool
  - removed openai_api_key param for tools

Text2sql tested on [airline-passenger-satisfaction](https://www.kaggle.com/datasets/teejmahal20/airline-passenger-satisfaction) dataset
Fixes #8648

## Type of change
- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/langchain_handler/README.md
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/langchain_handler/README.md:28]: @ea-rus 
This one should be `anyscale_endpoints_api_key`.

File: mindsdb/integrations/handlers/langchain_handler/README.md
Comment (martyna-mindsdb) [inline mindsdb/integrations/handlers/langchain_handler/README.md:1]: @ea-rus 
Please let me know once you merge this PR, so I'll update the docs.

File: mindsdb/integrations/handlers/langchain_handler/README.md
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/README.md:28]: provider name is used as part of the key
to use param name `anyscale_endpoints_api_key` we need also to change provider param in the same command:
provider='anyscale_endpoints'

Comment (martyna-mindsdb) [reply mindsdb/integrations/handlers/langchain_handler/README.md:28]: Let's change this name to anyscale_endpoints - so that it is consistent with [anyscale_endpoints_handler](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/anyscale_endpoints_handler/anyscale_endpoints_handler.py#L24).

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:33]: Curious as to why the Python REPL was removed? Too unstable?

File: mindsdb/interfaces/chatbot/memory.py
Comment (paxcema) [inline mindsdb/interfaces/chatbot/memory.py:63]: Just to double check: the method does not return anything now, is this intended?

File: mindsdb/integrations/handlers/langchain_handler/tools.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/tools.py:147]: We need to document this change in behavior once merged.

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/skill_tool.py:104]: Great refactor :) much cleaner!

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/skill_tool.py:54]: Perhaps document _why_ this would be preferred over `_make_text_to_sql_langchain_tools`.

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/skill_tool.py:24]: Leftover? If so, remove.

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:63]: Out of curiosity, why do `anyscale` and `litellm` not support `conversational-full`? AFAIK if a provider supports one, it should be able to support the other.

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:235]: nitpick: `elif` on these clauses

File: mindsdb/integrations/handlers/langchain_handler/mindsdb_database_agent.py
Comment (paxcema) [inline mindsdb/integrations/handlers/langchain_handler/mindsdb_database_agent.py:174]: Nice!

**Review by paxcema — CHANGES_REQUESTED** [review]
Looking good! Left a few comments.

File: mindsdb/integrations/handlers/langchain_handler/README.md
Comment (martyna-mindsdb) [reply mindsdb/integrations/handlers/langchain_handler/README.md:28]: @paxcema 
I discussed with @ea-rus offline that we'll keep anyscale_api_key name here because it doesn't use our handler but directly Anyscale API from LangChain.

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:235]: sure, even if they all have 'return', elif looks better
changed

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:63]: I think it should support the 'default' mode also. But I can't find where 'conversational-full' is used?

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/skill_tool.py:24]: done.
please take attention, this function will be removed in https://github.com/mindsdb/mindsdb/pull/8792

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/skill_tool.py:54]: the idea is: 
replace 4 tools generated by langchain (sql_db_query, sql_db_schema, sql_db_list_tables, sql_db_query_checker)
to one tool (sql_db_query) and, as we already know list of tables, provide all information about tables to use.

File: mindsdb/integrations/handlers/langchain_handler/tools.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/tools.py:147]: do you mean this item in PR description: 'don't use standard tools if skills are defined'?
@martyna-mindsdb, where is better to do it?

File: mindsdb/interfaces/chatbot/memory.py
Comment (ea-rus) [reply mindsdb/interfaces/chatbot/memory.py:63]: yes, it used to return value but the output isn't used anywhere

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (ea-rus) [reply mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:33]: It raises error on version =0.0.347: Got unknown tool python_repl
Looks like it[ was removed ](https://github.com/langchain-ai/langchain/blob/v0.0.347/libs/langchain/langchain/agents/load_tools.py#L99)due security issues

File: mindsdb/integrations/handlers/langchain_handler/skill_tool.py
Comment (paxcema) [reply mindsdb/integrations/handlers/langchain_handler/skill_tool.py:54]: Perfect. Then, let's add this to the docstring:

```
we replace all 4 tools with a single one (`sql_db_query`) and, as we already know the list of tables and their relevant information, we provide it directly instead.
```

File: mindsdb/integrations/handlers/langchain_handler/langchain_handler.py
Comment (paxcema) [reply mindsdb/integrations/handlers/langchain_handler/langchain_handler.py:63]: Ah, you may be right. I won't block the PR on this, feel free to keep it as-is.

**Review by paxcema — APPROVED** [review]

**Pull request merged**
